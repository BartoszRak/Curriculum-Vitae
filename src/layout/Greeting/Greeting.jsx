import React from 'react'
import { withStyles, Typography, Avatar } from '@material-ui/core'
import PropTypes from 'prop-types'

import backgroundUrl from '~assets/images/landscape-mountains-nature-dark.jpg'
import avatarUrl from '~assets/images/instagram-photo.jpg'

import style from './Greeting.style'

export function Greeting({ classes }) {
  return (
    <div
      className={classes.root}
      style={{
        backgroundImage: `url(${backgroundUrl})`,
      }}
    >
      <Avatar className={classes.avatar} src={avatarUrl} />
      <Typography variant="h1" className={classes.title}>
      Bartosz Rak
      </Typography>
      <Typography variant="h2" className={classes.subtitle}>
      Front-end developer
      </Typography>
    </div>
  )
}

Greeting.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Greeting)
