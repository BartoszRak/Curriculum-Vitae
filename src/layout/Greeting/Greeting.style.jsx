export default () => ({
  root: {
    alignItems: 'center',
    backgroundAttachment: 'fixed',
    display: 'flex',
    flexDirection: 'column',
    height: '100vh',
    justifyContent: 'center',
  },
})
