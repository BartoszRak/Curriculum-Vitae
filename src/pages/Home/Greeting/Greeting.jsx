import React from 'react'
import { FormattedMessage } from 'react-intl'
import { Typography, Avatar, Container } from '@material-ui/core'
import PropTypes from 'prop-types'

import avatarUrl from '~assets/images/instagram-photo2.jpg'

import useStyle from './Greeting.style'

export function Greeting({ classes: overridingClasses }) {
  const classes = { ...useStyle(), ...overridingClasses }
  return (
    <section className={classes.section}>
      <Container classes={{ root: classes.root }} id="greeting">
        <Avatar
          alt="Author instagram picture"
          className={classes.avatar}
          src={avatarUrl}
        />
        <Typography
          className={classes.title}
          color="secondary"
          data-testid="greeting-title"
          variant="h1"
        >
          Bartosz Rak
        </Typography>
        <Typography
          className={classes.subtitle}
          color="secondary"
          data-testid="greeting-subtitle"
          variant="h2"
        >
          <FormattedMessage id="pages.home.greeting.subtitle" />
          <span aria-hidden>üòÅ</span>
        </Typography>
        <Typography
          className={classes.subsubtitle}
          color="secondary"
          data-testid="greeting-subsubtitle"
          variant="h3"
        >
          <FormattedMessage id="pages.home.greeting.subsubtitle" />
        </Typography>
      </Container>
    </section>
  )
}

Greeting.propTypes = {
  classes: PropTypes.object,
}

export default Greeting
