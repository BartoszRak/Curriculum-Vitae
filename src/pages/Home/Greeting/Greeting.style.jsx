import { makeStyles } from '@material-ui/core'

export default makeStyles(({ breakpoints, palette, spacing }) => ({
  avatar: {
    border: `2px solid ${palette.secondary.main}`,
    height: 250,
    marginBottom: 60,
    width: 250,
  },
  root: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    width: '100%',
  },
  section: {},
  subsubtitle: {
    fontWeight: 'lighter',
    marginTop: spacing(0.5),
    textAlign: 'center',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('sm')]: {
      fontSize: '1.2rem',
    },
  },
  subtitle: {
    textAlign: 'center',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('sm')]: {
      fontSize: '1.6rem',
    },
  },
  title: {
    textAlign: 'center',
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('sm')]: {
      fontSize: '2.5rem',
    },
  },
}))
