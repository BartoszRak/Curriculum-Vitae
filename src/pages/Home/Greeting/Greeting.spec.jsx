import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { Greeting } from './Greeting'

jest.unmock('@material-ui/core')

jest.mock('react-intl')

describe('component: Greeting', () => {
  afterEach(cleanup)

  describe('rendering', () => {
    test('render without crash ', () => {
      const wrapper = render(<Greeting />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = render(<Greeting />)

      expect(wrapper).toMatchSnapshot()
    })

    describe('render all titles', () => {
      test.each([
        ['greeting-title'],
        ['greeting-subtitle'],
        ['greeting-subsubtitle']
      ])('when specified title %s provided', title => {
        const { getByTestId } = render(<Greeting />)
        const titleEl = getByTestId(title)

        expect(titleEl).toBeVisible()
      })
    })
  })
})
