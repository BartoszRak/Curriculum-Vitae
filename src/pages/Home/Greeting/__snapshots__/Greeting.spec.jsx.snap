// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Greeting rendering match snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="makeStyles-section-49"
      >
        <div
          class="MuiContainer-root makeStyles-root-48 MuiContainer-maxWidthLg"
          id="projects"
        >
          <div
            class="MuiAvatar-root makeStyles-avatar-47"
          >
            <img
              alt="Author instagram picture"
              class="MuiAvatar-img"
              src="instagram-photo2.jpg"
            />
          </div>
          <h1
            class="MuiTypography-root MuiTypography-h1 MuiTypography-colorSecondary makeStyles-title-52"
            data-testid="greeting-title"
          >
            Bartosz Rak
          </h1>
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-colorSecondary makeStyles-subtitle-51"
            data-testid="greeting-subtitle"
          >
            <span
              aria-hidden="true"
            >
              üòÅ
            </span>
          </h2>
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-colorSecondary makeStyles-subsubtitle-50"
            data-testid="greeting-subsubtitle"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="makeStyles-section-49"
    >
      <div
        class="MuiContainer-root makeStyles-root-48 MuiContainer-maxWidthLg"
        id="projects"
      >
        <div
          class="MuiAvatar-root makeStyles-avatar-47"
        >
          <img
            alt="Author instagram picture"
            class="MuiAvatar-img"
            src="instagram-photo2.jpg"
          />
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h1 MuiTypography-colorSecondary makeStyles-title-52"
          data-testid="greeting-title"
        >
          Bartosz Rak
        </h1>
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-colorSecondary makeStyles-subtitle-51"
          data-testid="greeting-subtitle"
        >
          <span
            aria-hidden="true"
          >
            üòÅ
          </span>
        </h2>
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-colorSecondary makeStyles-subsubtitle-50"
          data-testid="greeting-subsubtitle"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
